app/services/migration/migrators/contract.rb

```diff
@@ -0,0 +1,67 @@
+module Migration::Migrators
+  class Contract < Base
+    class << self
+      def record_count
+        ecf_contracts.count
+      end
+
+      def model
+        :contract
+      end
+
+      def ecf_contracts
+        Migration::Ecf::NpqContract.includes(:cohort, npq_lead_provider: [:cpd_lead_provider])
+      end
+
+      def dependencies
+        %i[cohort lead_provider course statement]
+      end
+    end
+
+    def call
+      migrate(self.class.ecf_contracts) do |ecf_contract|
+        course = courses_by_identifier[ecf_contract.course_identifier]
+
+        ecf_statements(ecf_contract).find_each do |ecf_statement|
+          statement = ::Statement.find_by!(ecf_id: ecf_statement.id)
+
+          contract_template = ::ContractTemplate.find_or_initialize_by(ecf_id: ecf_contract.id)
```

Comment:
As the attributes have the same names could do:


```
    SHARED_ATTRIBUTES = %w[
      created_at
      updated_at
      service_fee_percentage
      output_payment_percentage
      per_participant
      number_of_payment_periods
      recruitment_target
      service_fee_installments
      targeted_delivery_funding_per_participant
      monthly_service_fee
      special_course
    ].freeze

contract_template.update!(ecf_contract.attributes.slice(SHARED_ATTRIBUTES))
```

Or if there are only a few we don't want to migrate across, something like:

```
contract_template.update!(ecf_contract.attributes.excluding(:id))
```